health/micro-ui/web/micro-ui-internals/packages/modules/campaign-manager/src/components/fetchFromMicroplan.js

```
@@ -110,21 +100,25 @@ const FetchFromMicroplanScreen = () => {
+        // Show loading state
+        setShowToast({ key: "info", label: t("FETCHING_MICROPLAN_DATA") });
+
+        setTimeout(() => {
+          // Ensure progressive updates
+          setCurrentStep((prev) => MICROPLAN_FETCH_STEP);
+        }, MICROPLAN_FETCH_TIMEOUT);
       } catch (error) {
         console.error("Error fetching microplan data:", error);
       }
     };
     if (templates && templates?.completed) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)